src/pipdeptree/_models/dag.py

```diff
@@ -65,7 +63,8 @@ def from_pkgs(cls, pkgs: list[Distribution]) -> PackageDAG:
                     req = next(requires_iterator)
                 except InvalidRequirementError as err:
                     # We can't work with invalid requirement strings. Let's warn the user about them.
-                    dist_name_to_invalid_reqs_dict.setdefault(p.project_name, []).append(str(err))
+                    if warning_printer.should_warn():
```

Comment:
My reasoning behind this was that users who pass --warn silence would be building these data structures that would end up never being used since warnings can't be printed. However, it also feels weird having to keep performing this check even though its return value should never change.

Reply:
You're right, if we don't check here, this part of the data structures would be wasted and there would be unnecessary time consumption. It seems better to keep the current situation.